lib/MySQL_Catalog.cpp

```
@@ -0,0 +1,356 @@
+		"  links = ?5,"
+		"  updated_at = strftime('%s', 'now')";
+
+	int rc = db->prepare_v2(upsert_sql, &stmt);
+	if (rc != SQLITE_OK) {
+		proxy_error("Failed to prepare catalog upsert: %d\n", rc);
+		return -1;
+	}
+
+	(*proxy_sqlite3_bind_text)(stmt, 1, kind.c_str(), -1, SQLITE_TRANSIENT);
+	(*proxy_sqlite3_bind_text)(stmt, 2, key.c_str(), -1, SQLITE_TRANSIENT);
+	(*proxy_sqlite3_bind_text)(stmt, 3, document.c_str(), -1, SQLITE_TRANSIENT);
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)